packages/menu/src/Menu/Menu.tsx

```diff
@@ -100,7 +101,7 @@ export const Menu = React.forwardRef<HTMLDivElement, MenuProps>(function Menu(
 
   const [, setClosed] = useState(false);
   const currentSubMenuRef = useRef<SubMenuType | null>(null);
-  const [uncontrolledOpen, uncontrolledSetOpen] = useState(false);
+  const [uncontrolledOpen, uncontrolledSetOpen] = useState(initialOpen);
```

Comment:
We should consider using `useControlledValue` for  `initialOpen`, `isOpen` & `setOpen`